### DISCUSSION TITLE
How to build pydantic docs to be used in offline env?

### DISCUSSION THREAD

**Discussion by honglei** [pydantic/pydantic#5423] [Question] [answered]
Env: Win10/python3.11.3/pydantic-1.10.7
I've produced `E:\xx\libs\pydantic-1.10.7\docs\.tmp_examples` by `docs\build\main.py`
build html by cmd :`c:\python311\python.exe -m mkdocs build -f mkdocs.yml -d ./output`

but these md files in `.tmp_examples` not include in html pages. 

![image](https://user-images.githubusercontent.com/1011330/230852354-64b9273c-c071-4174-917b-5f88120859ee.png)

**Comment by samuelcolvin** [comment]
Looks like you're building old v1.10 docs, you need to run `make docs` to build the examples.

For V2 just `mkdocs build` should work, but the V2 docs are very much a WIP.

**Reply by honglei** [reply, marked as answer]
The problem may caused by the Windows system, I swiched to Debian10, and all works fine.

**Reply by honglei** [reply]
Step1:
Debian10 ,run  `docs/build/main.py` build `.tmp_examples`   
Setp2:
use the result of Step1 in Win10 env , run `mkdocs build`
The output works well, so the `docs/build/*.py`  has problem under win10. 

##Problem1:

should  use encoding='utf-8' explicitly  under Windows, otherwise it will cause `builtins.UnicodeDecodeError: 'gbk' codec can't decode byte 0xa6 in position 29050: illegal multibyte sequence`:

main.py
```python
    history = (PROJECT_ROOT / 'HISTORY.md').read_text(encoding='utf-8')
    #.... 
    (PROJECT_ROOT / 'docs/.changelog.md').write_text(history, encoding='utf-8')
```
